File: n2/2.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Введите кол-во полных банок и эквивалент обмена: ");
        int[] cups = { sc.nextInt(), sc.nextInt() };

        int drunk = cups[0]; // начальное количество выпитых банок
        int i = 0; // счётчик итераций
        int full = 0; // количество банок, полученных при обмене

        // пока можно обменять пустые банки на полные
        while (cups[0] >= cups[1]) {
            i += 1; // шаг: опустошение банок
            full = cups[0] / cups[1]; // обмен: сколько полных получаем
            i += 1; // шаг: сам обмен
            int empty = cups[0] % cups[1]; // остаток банок, которые не удалось обменять
            drunk += full; // прибавляем полученные к выпитым
            cups[0] = full + empty; // обновляем количество банок (новые + оставшиеся)
        }

        i += 1; // шаг: выпиваем всё, что осталось
        System.out.println("Выпито банок и кол-во итераций: " + drunk + " " + i);
    }
}

File: n2/2.php
<?php
echo "Введите кол-во полных банок и эквивалент обмена: ";
$cups = array_map('intval', explode(" ", trim(fgets(STDIN))));

$drunk = $cups[0]; // изначально выпито
$i = 0;            // счётчик шагов
$full = 0;         // количество банок после обмена
if ($cups[0] > 1){
    // пока можно менять
    while ($cups[0] >= $cups[1]) {
        $i += 1; // шаг: опустошили
        $full = intdiv($cups[0], $cups[1]); // обмен
        $i += 1;
        $empty = $cups[0] % $cups[1]; // остаток
        $drunk += $full;
        $cups[0] = $full + $empty; // обновляем запас
    }
    
    $i += 1; // допиваем всё
    echo "Выпито банок и кол-во итераций: $drunk $i\n";
}
?>
